pkg/controller/plan/kubevirt.go

```diff
@@ -859,6 +891,105 @@ func (r *KubeVirt) GetGuestConversionPod(vm *plan.VMStatus) (pod *core.Pod, err
 	return
 }
 
+func (r *KubeVirt) GetVirtV2VConvertedVMConfig(vm *plan.VMStatus, pod *core.Pod, step *plan.Step) (err error) {
+	if pod.Status.PodIP == "" {
+		//we need the IP for fetching the firmware info.
+		return
+	}
+
+	url := fmt.Sprintf("http://%s:8080/xml", pod.Status.PodIP)
+	maxRetries := 3
+	retryInterval := time.Second * 5
+
+	resp, err := http.Get(url)
+	if err == nil {
+		defer resp.Body.Close()
+	} else {
+		if strings.Contains(err.Error(), "connection refused") {
+			return nil
+		} else {
+			for i := 0; i < maxRetries; i++ {
+				resp, err = http.Get(url)
+				if err != nil {
+					if i < maxRetries-1 {
+						fmt.Printf("Retrying... (%d/%d)\n", i+1, maxRetries)
+						defer resp.Body.Close()
+						time.Sleep(retryInterval)
+						continue
+					}
+				}
+				defer resp.Body.Close()
+				break
+			}
+			// Handle the final error after retries
+			if err != nil {
+				return
+			}
+		}
+	}
+
+	vmConfigXML, errFirrmware := io.ReadAll(resp.Body)
+	if errFirrmware != nil {
+		return errFirrmware
+	}
+	firmware, err := getFirmwareFromConfig(string(vmConfigXML))
+
+	currentAnnotations := r.Migration.Annotations
+	if currentAnnotations == nil {
+		currentAnnotations = make(map[string]string)
+	}
+	var vmId string
+	labels := pod.GetLabels()
+	if val, ok := labels["vmID"]; ok {
+		vmId = val
+	} else {
+		return errors.New("failed to get virt-v2v pod labels")
+	}
+
+	currentAnnotations[vmId] = firmware
+	r.Migration.SetAnnotations(currentAnnotations)
+	err = r.Destination.Client.Update(context.TODO(), r.Migration.DeepCopy())
+	if err != nil {
+		return
+	}
+
+	shutdownURL := fmt.Sprintf("http://%s:8080/shutdown", pod.Status.PodIP)
+	resp, err = http.Get(shutdownURL)
+	if err != nil {
+		return
+	}
+	defer resp.Body.Close()
+
+	return nil
+}
+
+func readConfFromXML(xmlData string) (*Domain, error) {
+	var domain Domain
+
+	reader := strings.NewReader(xmlData)
+	decoder := xml.NewDecoder(reader)
+
+	err := decoder.Decode(&domain)
+	if err != nil {
+		return &domain, err
+	}
+	fmt.Printf("this is domain %v", domain)
```

Comment:
forgotten? or it should be logged? if log, i would expect the usage of `r.Log` and more details (vm and so on)